src/System.IO/src/System/IO/BufferedStream.cs

```diff
@@ -234,7 +236,11 @@ protected override void Dispose(bool disposing)
             finally
             {
                 _stream = null;
-                _buffer = null;
+                if (_buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
Yes, this isn't about complete security; after all, this all also in the same process, in the same app domain, code could easily read from whatever memory it wanted to, etc.  This is more about minimizing unexpected consequences.